### PR TITLE
Improve the info provided by listModelsWithInfo.

### PR DESCRIPTION
**Pull request by frankban**
Also use this additional info when listing the models in the profile view.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6752/

**Comment by frankban** [comment]
**QA** must be done in both gijoe and gisf scenarios.
For gisf, at this stage you can use `guiproxy -config 'gisf: true, jujuEnvUUID: ""'`.

Open the profile view and check that:
- information is now displayed for each model, including cloud/region, last connection (in human friendly form), number of machines and model owner;
- you can still switch to models and/or delete models.

File: jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js [hunk 1]
```diff
@@ -259,7 +259,11 @@ YUI.add('user-profile-model-list', function() {
           return null;
         }
       }
-      const lastConnection = model.lastConnection || '--';
+      const region = model.region || 'no region';
+      let owner = '--';
```
Comment (mitechie) [inline jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js]: can we not have an owner? should this be some sort of warning/error condition?

File: jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js [hunk 2]
```diff
@@ -272,18 +276,18 @@ YUI.add('user-profile-model-list', function() {
...
           </span>
           <span className="user-profile__list-col two-col">
             <juju.components.DateDisplay
-              date={lastConnection}
+              date={model.lastConnection || '--'}
               relative={true} />
           </span>
           <span className="user-profile__list-col one-col">
-            --
+            {model.numMachines}
```
Comment (mitechie) [inline jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js]: does this come back as 0 for new models?

File: jujugui/static/gui/src/app/store/env/controller-api.js [hunk 3]
```diff
@@ -509,11 +560,25 @@ YUI.add('juju-controller-api', function(Y) {
...
+          - access: the type of access the user has as a string, like "read" or
+            "admin";
+          - err: a message describing a specific user error, or undefined;
         - life: the lifecycle status of the model: "alive", "dying" or "dead";
         - isAlive: whether the model is alive or dying/dead;
         - isController: whether the model is a controller model;
-        - lastConnection: the date of the last connection as a string, e.g.:
-          '2015-09-24T10:08:50Z' or null if the model was never connected to;
+        - lastConnection: the date of the last connection of the current user
+          as a Date object, or null if the model was never connected to;
```
Comment (kadams54) [inline jujugui/static/gui/src/app/store/env/controller-api.js]: Thanks for standarizing this as a Date object.

File: jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js (hunk 1 above)
Comment (frankban) [inline jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js]: Agreed, we should always have an owner. The defensiveness is due to pre-existing code. I'll add a comment.

File: jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js (hunk 2 above)
Comment (frankban) [inline jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js]: Yes it does.

**Comment by kadams54** [comment]
QA is good in `gisf`. Checking `gijoe`.

**Comment by kadams54** [comment]
`gijoe` also looks good. Thanks @frankban!

**Comment by frankban** [comment]
Thanks all for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6756/

**Pull request merged**